spring-integration-jmx/src/main/java/org/springframework/integration/monitor/IntegrationMBeanExporter.java

```diff
@@ -250,24 +250,22 @@ public Object postProcessAfterInitialization(Object bean, String beanName) throw
 			return bean;
 		}
 
-		if (bean instanceof MessageHandler) {
+		if (bean instanceof MessageHandlerMetrics) {
 			if (this.handlerInAnonymousWrapper(bean) != null) {
 				if (logger.isDebugEnabled()) {
 					logger.debug("Skipping " + beanName + " because it wraps another handler");
 				}
 				return bean;
 			}
-			SimpleMessageHandlerMetrics monitor = new SimpleMessageHandlerMetrics((MessageHandler) bean);
-			Object advised = applyHandlerInterceptor(bean, monitor, beanClassLoader);
+			MessageHandlerMetrics monitor = (MessageHandlerMetrics) extractTarget(bean);
```

Comment:
Good point - I was trying to be safe in case something else had advised it, but since we only reference interfaces, it's not necessary.